Source/santametricservice/Writers/SNTMetricHTTPWriterTest.m

```diff
@@ -0,0 +1,144 @@
+#import <Foundation/Foundation.h>
+#import <XCTest/XCTest.h>
+
+#import <MOLAuthenticatingURLSession/MOLAuthenticatingURLSession.h>
+#import <OCMock/OCMock.h>
+
+#import "Source/santametricservice/Writers/SNTMetricHTTPWriter.h"
+
+@interface SNTMetricHTTPWriterTest : XCTestCase
+@property id mockSession;
+@property id mockSessionDataTask;
+@property id mockMOLAuthenticatingURLSession;
+@property SNTMetricHTTPWriter *httpWriter;
+@end
+
+@implementation SNTMetricHTTPWriterTest
+
+- (void)setUp {
+  self.mockSession = [OCMockObject niceMockForClass:[NSURLSession class]];
+  self.mockSessionDataTask = [OCMockObject niceMockForClass:[NSURLSessionDataTask class]];
+  self.mockMOLAuthenticatingURLSession =
+    [OCMockObject niceMockForClass:[MOLAuthenticatingURLSession class]];
+  [[[self.mockMOLAuthenticatingURLSession stub] andReturn:self.mockMOLAuthenticatingURLSession]
+    alloc];
+  [[[self.mockMOLAuthenticatingURLSession stub] andReturn:self.mockSession] session];
+
+  self.httpWriter = [[SNTMetricHTTPWriter alloc] init];
+}
+
+- (void)tearDown {
+  [self.mockSessionDataTask stopMocking];
+  [self.mockSession stopMocking];
+  [self.mockMOLAuthenticatingURLSession stopMocking];
+}
+
+- (void)createMockResponseWithURL:(NSURL *)url
+                         withCode:(NSInteger)code
+                         withData:(NSData *)data
+                        withError:(NSError *)err {
+  NSHTTPURLResponse *response =
+    [[NSHTTPURLResponse alloc] initWithURL:url
+                                statusCode:code
+                               HTTPVersion:@"HTTP/1.1"
+                              headerFields:@{@"content-type" : @"application/json"}];
+
+  __block void (^passedBlock)(NSData *, NSURLResponse *, NSError *);
+
+  void (^getCompletionHandler)(NSInvocation *) = ^(NSInvocation *invocation) {
+    [invocation getArgument:&passedBlock atIndex:3];
+  };
+
+  void (^callCompletionHandler)(NSInvocation *) = ^(NSInvocation *invocation) {
+    passedBlock(data, response, err);
+  };
+
+  // stub out session to call completion handler immediately.
+  [(NSURLSessionDataTask *)[[self.mockSessionDataTask stub] andDo:callCompletionHandler] resume];
+
+  // stub out NSURLSession to assign our completion handler and return our mock
+  [[[[self.mockSession stub] andDo:getCompletionHandler] andReturn:self.mockSessionDataTask]
+    dataTaskWithRequest:[OCMArg any]
+      completionHandler:[OCMArg any]];
+}
+
+- (void)testValidPostOfData {
+  NSURL *url = [[NSURL alloc] initWithString:@"http://localhost:8444/submit"];
+
+  [self createMockResponseWithURL:url withCode:200 withData:nil withError:nil];
+
+  SNTMetricHTTPWriter *httpWriter = [[SNTMetricHTTPWriter alloc] init];
+
+  NSData *JSONdata = [@"{\"foo\": \"bar\"}\r\n" dataUsingEncoding:NSUTF8StringEncoding];
+
+  NSError *err;
+  BOOL result = [httpWriter write:@[ JSONdata ] toURL:url error:&err];
+  XCTAssertEqual(YES, result);
+  XCTAssertNil(err);
+}
```

Comment:
Also add test/assert on the response handler here to check if the data is still intact when exported?